hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -186,6 +206,71 @@ public static DBCheckpoint createOmSnapshotCheckpoint(
     return dbCheckpoint;
   }
 
+  /**
+   * Helper method to delete keys in the snapshot scope from active DB's
+   * deletedTable.
+   *
+   * @param omMetadataManager OMMetadataManager instance
+   * @param volumeName volume name
+   * @param bucketName bucket name
+   */
+  private static void deleteKeysInSnapshotScopeFromDTableInternal(
+      OMMetadataManager omMetadataManager,
+      String volumeName,
+      String bucketName) throws IOException {
+
+    // Range delete start key (inclusive)
+    String beginKey =
+        omMetadataManager.getOzoneKey(volumeName, bucketName, "");
+
+    // Range delete end key (exclusive) to be found
+    String endKey;
+
+    try (TableIterator<String,
+        ? extends Table.KeyValue<String, RepeatedOmKeyInfo>>
+        keyIter = omMetadataManager.getDeletedTable().iterator()) {
+
+      // TODO: Start timer for perf tracking
+      keyIter.seek(beginKey);
+      // Continue only when there are entries of snapshot (bucket) scope
+      // in deletedTable in the first place
+      if (!keyIter.hasNext()) {
+        // Use null as a marker. No need to do deleteRange() at all.
+        endKey = null;
+      } else {
+        // Remember the last key with a matching prefix
+        endKey = keyIter.next().getKey();
+
+        // Loop until prefix mismatches.
+        // TODO: Room for optimization? Seek next predicted bucket name.
+        while (keyIter.hasNext()) {
```

Comment:
why don't we use  keyIter.seekToLast() ?

Reply:
`seekToLast()` doesn't take a param. It goes to the last entry in the table, not the last entry that matches the given prefix.